sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -664,8 +666,9 @@ def read(self, size: int = -1, *, chars: Optional[int] = None) -> T:
 
         # Start by reading from current_content
         start = self._current_content_offset
-        length = min(len(self._current_content) - self._current_content_offset, size - count)
-        read = output_stream.write(self._current_content[start:start + length])
+        length = min(len(self._current_content) - self._current_content_offset,
+                     size - count)
+        read = output_stream.write(self._current_content[start:start + length])  # type: ignore [arg-type]
```

Comment:
azure\storage\blob\_download.py:669: error: Argument 1 to "write" of "BufferedIOBase" has incompatible type "Union[str, bytes]"; expected "Buffer"  [arg-type]
azure\storage\blob\_download.py:669: error: Argument 1 to "write" of "TextIOBase" has incompatible type "Union[str, bytes]"; expected "str"  [arg-type]

MyPy is not able to tell since there's a 2 jump here (type of `output_stream`, type of `self._current_content`)